### ISSUE TITLE
Lua Error

### ISSUE THREAD

**Issue by rainfordays** [typicalzergling/vendor#40]
Happens when mousing over an item [Hardened Molars](gray qual) after moving it in inventory.

It keeps happening for a few seconds then it "settles" and everything works correctly. Not sure if it's tied to building itemcache or something but now you know.

```
Message: Interface\AddOns\Vendor\ui\tooltip.lua:98: attempt to compare number with table
Time: Sat Oct 17 12:24:20 2020
Count: 3
Stack: Interface\AddOns\Vendor\ui\tooltip.lua:98: attempt to compare number with table
[string "@Interface\AddOns\TradeSkillMaster\LibTSM\Service\ErrorHandler.lua"]:758: in function <...Ons\TradeSkillMaster\LibTSM\Service\ErrorHandler.lua:717>
[string "=[C]"]: ?
[string "@Interface\AddOns\Vendor\ui\tooltip.lua"]:98: in function `AddItemTooltipLines'
[string "@Interface\AddOns\Vendor\ui\tooltip.lua"]:36: in function `?'
[string "@Interface\AddOns\Vendor\sys\hook.lua"]:42: in function <Interface\AddOns\Vendor\sys\hook.lua:42>
[string "@Interface\AddOns\Vendor\sys\hook.lua"]:52: in function <Interface\AddOns\Vendor\sys\hook.lua:50>
[string "=[C]"]: ?
[string "=[C]"]: ?
[string "=[C]"]: ?
[string "@Interface\AddOns\TradeSkillMaster\LibTSM\Service\ItemTooltipClasses\Wrapper.lua"]:98: in function <...Master\LibTSM\Service\ItemTooltipClasses\Wrapper.lua:96>
[string "=[C]"]: in function `SetBagItem'
[string "@Interface\FrameXML\ContainerFrame.lua"]:1321: in function `ContainerFrameItemButton_OnUpdate'
[string "@Interface\FrameXML\ContainerFrame.lua"]:1368: in function `ContainerFrameItemButton_OnEnter'
[string "*:OnEnter"]:1: in function <[string "*:OnEnter"]:1>
[string "=[C]"]: ?

Locals: errMsg = "Interface\AddOns\Vendor\ui\tooltip.lua:98: attempt to compare number with table"
isBugGrabber = nil
tsmErrMsg = nil
oldModule = nil
private = <table> {
 globalNameTranslation = <table> {
 }
 FormatErrorMessageSection = <function> defined @Interface\AddOns\TradeSkillMaster\LibTSM\Service\ErrorHandler.lua:517
 ErrorHandler = <function> defined @Interface\AddOns\TradeSkillMaster\LibTSM\Service\ErrorHandler.lua:134
 origErrorHandler = <function> defined @Interface\SharedXML\SharedBasicControls.lua:355
 hitInternalError = false
 CreateErrorFrame = <function> defined @Interface\AddOns\TradeSkillMaster\LibTSM\Service\ErrorHandler.lua:536
 num = 0
 SanitizeString = <function> defined @Interface\AddOns\TradeSkillMaster\LibTSM\Service\ErrorHandler.lua:510
 AddonBlockedHandler = <function> defined @Interface\AddOns\TradeSkillMaster\LibTSM\Service\ErrorHandler.lua:502
 GetStackLevelInfo = <function> defined @Interface\AddOns\TradeSkillMaster\LibTSM\Service\ErrorHandler.lua:344
 ParseLocals = <function> defined @Interface\AddOns\TradeSkillMaster\LibTSM\Service\ErrorHandler.lua:396
 ignoreErrors = false
 errorReports = <table> {
 }
 IsTSMAddon = <function> defined @Interface\AddOns\TradeSkillMaster\LibTSM\Service\ErrorHandler.lua:478
 localLinesTemp = <table> {
 }
 GetStackInfo = <function> defined @Interface\AddOns\TradeSkillMaster\LibTSM\Service\ErrorHandler.lua:313
}
MAX_STACK_DEPTH = 50
OLD_TSM_MODULES = <table> {
 1 = "TradeSkillMaster_Accounting"
 2 = "TradeSkillMaster_AuctionDB"
 3 = "TradeSkillMaster_Auctioning"
 4 = "TradeSkillMaster_Crafting"
 5 = "TradeSkillMaster_Destroying"
 6 = "TradeSkillMaster_Mailing"
 7 = "TradeSkillMaster_Shopping"
 8 = "TradeSkillMaster_Vendoring"
 9 = "TradeSkillMaster_Warehousing"
}

```

**Comment by typicalzergling** [comment]
Does this still repro on 4.1.3? I think this is fixed now.

**Comment by rainfordays** [comment]
Yes, im guessing its something to do with not resetting tooltip or caching it because it works alright if I hover over another item.

Like this:

Move gray item to another bagslot, hover over gray item produces lua error.
Move gray item to another bagslot, hover another item and then the gray item again works flawlessly.

**Comment by typicalzergling** [comment]
Fixed in 4.1.4
